src/vs/editor/contrib/rename/rename.ts

```diff
@@ -173,14 +173,15 @@ class RenameController implements IEditorContribution {
 		let selection = this.editor.getSelection();
 		let selectionStart = 0;
 		let selectionEnd = loc.text.length;
+		(this._cts as EditorStateCancellationTokenSource).range = loc.range;
```

Comment:
This is not nice. The `range` property should not be mutable as that counters the idea of capturing a state to compare it against a future state. How about creating a new token?  E.g. the token from line 146 is used to resolve the rename locations, then here dispose it and create new token with a (readonly) range

Reply:
Done!